librz/core/cmd/cmd_analysis.c

```diff
@@ -1311,20 +1302,15 @@ RZ_API bool rz_core_esil_continue_back(RzCore *core) {
 	return true;
 }
 
-static void cmd_address_info(RzCore *core, const char *addrstr, int fmt) {
+static void cmd_address_info(RzCore *core, const char *addrstr, PJ *pj) {
```

Comment:
pass RzCmdStateOutput to this function and use `ut64 addr` instead of doing the conversion to num here. The decision of whether to print JSON or not should be done on state->mode and not on whether `pj` is NULL or not, because state->pj is in a union, so it could be != NULL even when JSON is not selected.